src/queries/useGnosisBatch.tsx

```diff
@@ -0,0 +1,46 @@
+// source: https://github.com/LlamaPay/interface/blob/main/queries/useGnosisBatch.tsx 🧠
+import SafeAppsSDK from "@gnosis.pm/safe-apps-sdk/dist/src/sdk";
+import { useMutation } from "@tanstack/react-query";
+
+interface IUseGnosisBatch {
+  calls: { [key: string]: string[] };
+}
+
+interface IGnosisBatch extends IUseGnosisBatch {
+  sdk?: SafeAppsSDK;
+}
+
+async function batchCall({ sdk, calls }: IGnosisBatch) {
+  try {
+    if (!sdk) {
+      throw new Error("Couldn't get SDK");
+    } else {
+      const transactions: {
+        to: string;
+        value: string;
+        data: string;
+      }[] = [];
+
+      Object.keys(calls).forEach((p) => {
+        calls[p].forEach((q) => {
+          transactions.push({
+            to: p,
+            value: "0",
```

Comment:
it could be batched also (native tx or tx which needs to send eth internally), i opted to keep at zero for now, since did not see an use-case at first glance in smart-garden

but would say that whenever the use-case arises is a simple reading the value from `calls` and use it there

Reply:
ah i was for example thinking about the staker plugin sending eth and velo to deposit, and then the stake thereof all in the same batch?

Reply:
ah that is true, guess this worry does not need a solution any time soon then :)